content/ide-integration/_index.md

```diff
@@ -3,15 +3,120 @@ title: "IDE Integration"
 weight: 8
 ---
 
+
+## Downloading the source
+
+Either git clone the tinygo source from.
+
+> https://github.com/tinygo-org/tinygo
+
+## Go mod init in your project
+
+Navigate to your project and type:
+
+> go mod init
+
+## Replace needed imports
+
+Replace each std package you need using the replace keyword in the go.mod file.
+
+So if you want the machine package to be resolved replace it using.
+
+> replace machine => /path/to/tinygo/machine
+
+Repeat this step for every package, which needs to be resolved in your project.
+
+### Example
+
+```go
+module github.com/Nerzal/tinygo-playground
+
+go 1.14
+
+replace machine => /home/tobias/go/src/github.com/tinygo-org/tinygo/src/machine
+
+require (
+    machine v0.0.0-00010101000000-000000000000
+    tinygo.org/x/drivers v0.13.0
+)
+
+```
+
+## Add go module files to the packages
+
+In order to get your package resolved, you also have to add a go mod file for the packages.
+
+So navigate to `/path/to/tinygo/src`
+And then do `go mod init`
+
+### Example
+
+```bash
+ cd ~/go/src/github.com/tinygo-org/tinygo/src/machine
+ go mod init
+```
+
 ## GoPath
 
 In order to get IDE support like autocomplete you have to add the tinygo src path to your [GOPATH](https://github.com/golang/go/wiki/GOPATH).
 After adding the tinygo source path to the GOPATH go will know, where to look for the packages like `machine`
 
 ### Ubuntu Example
+
 You can just append the path to your tinygo installation in your GOPATH.
 
->  export GOPATH=$GOPATH:/path/to/your/tinygo
+> export GOPATH=$GOPATH:/path/to/your/tinygo
```

Comment:
At this point, if you are in src/examples/bliny1/blinky.go, machine.Output can be handled by gopls.
You may want to prompt them to check it once.

At this point, machine.LED cannot be processed by gopls.
This is because you do not have build-tag or other settings in place.

Reply:
i just added that comment :)

Reply:
thanks